engine/crates/runtime/src/cache/cached.rs

```diff
@@ -0,0 +1,895 @@
+use std::fmt::Display;
+use std::future::Future;
+use std::sync::Arc;
+
+use futures_util::{FutureExt, TryFutureExt};
+use tracing_futures::Instrument;
+
+use crate::cache::{
+    Cache, CacheReadStatus, Cacheable, CachedExecutionResponse, Entry, EntryState, GlobalCacheConfig,
+    RequestCacheConfig,
+};
+use crate::context::RequestContext;
+
+#[derive(Debug, PartialEq, Eq)]
+enum CacheResponse<T> {
+    Hit(T),
+    Miss(T),
+    Bypass(T),
+    Stale { response: T, updating: bool },
+}
+
+impl<T: Cacheable> From<CacheResponse<Arc<T>>> for CachedExecutionResponse<Arc<T>> {
```

Comment:
Just asking, but why do we need both `CacheResponse` and `CachedExecutionResponse`?  They seem very similar just from glancing at this code.

Reply:
You're correct, there _was_ a reason but completely unnecessary.